flow/src/mock.rs

```diff
@@ -4,33 +4,46 @@ pub use super::*;
 use frame_support::{
 	construct_runtime, parameter_types, PalletId,
 	traits::{Everything, GenesisBuild, Nothing},
+	pallet_prelude::*,
 };
 use sp_core::H256;
 use sp_runtime::{traits::IdentityLookup, Permill};
 
 use orml_traits::parameter_type_with_key;
 
-impl Campaign<Hash, AccountId, Balance, BlockNumber, Moment> {
-	pub fn new(campaign_id: Hash, expiry: BlockNumber) -> Campaign<Hash, AccountId, Balance, BlockNumber, Moment> {
+impl Campaign<Hash, AccountId, Balance, BlockNumber, Moment, BoundedVec<u8, <Test as Config>::StringLimit>> {
+	pub fn new(campaign_id: Hash, expiry: BlockNumber) -> Campaign<Hash, AccountId, Balance, BlockNumber, Moment, BoundedVec<u8, <Test as Config>::StringLimit>> {
 		Campaign {
 			id: campaign_id,
 			org: H256::random(),
-			name: vec![1, 2],
+			name: BoundedVec::truncate_from(vec![1, 2]),
```

Comment:
afair we keep names off chain or do we need this for the slug/reservation later on?

Reply:
True, name is stored off chain. It will be dropped in next updates. For now it exists for backwards compatibility.